File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeNthFromEnd(ListNode* head, int n) {
        if (head->next == nullptr) {
            return nullptr;
        }
        ListNode* meow = new ListNode(0, head);
        ListNode* prev = meow;
        ListNode* cur = meow;

        for (int i = 0; i < n; i++) {
            cur = cur->next;
        }

        while (cur->next) {
            cur = cur->next;
            prev = prev->next;
        }

        prev->next = prev->next->next;

        return meow->next;

    }
};

File: 0019-remove-nth-node-from-end-of-list/0019-remove-nth-node-from-end-of-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeNthFromEnd(self, head: Optional[ListNode], n: int) -> Optional[ListNode]:
        if head == None or head.next == None:
            return None
        prev: ListNode = None
        cur: ListNode = head
        ret = ListNode(0, head)

        for i in range(n):
            cur = cur.next
        prev = ret
        while cur != None:
            cur = cur.next
            prev = prev.next
        prev.next = prev.next.next

        return ret.next